File: aoc2019/aoc2019-java/src/test/java/de/havox_design/aoc2019/day18/Day18Test.java
package de.havox_design.aoc2019.day18;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import java.util.stream.Stream;

class Day18Test {

    @ParameterizedTest
    @MethodSource("getDataForTask1")
    void testTask1(String fileName, long expectation) {
        Assertions.assertEquals(expectation, ManyWorldsInterpretation.processTask1(fileName));
    }

    private static Stream<Arguments> getDataForTask1() {
        return Stream.of(
                Arguments.of("de/havox_design/aoc2019/day18/day18part1sample1.txt", 8L),
                Arguments.of("de/havox_design/aoc2019/day18/day18part1sample2.txt", 86L),
                Arguments.of("de/havox_design/aoc2019/day18/day18part1sample3.txt", 132L),
                Arguments.of("de/havox_design/aoc2019/day18/day18part1sample4.txt", 136L),
                Arguments.of("de/havox_design/aoc2019/day18/day18part1sample5.txt", 81L)
        );
    }

    @ParameterizedTest
    @MethodSource("getDataForTask2")
    void testTask2(String fileName, long expectation) {
        Assertions.assertEquals(expectation, ManyWorldsInterpretation.processTask2(fileName));
    }

    private static Stream<Arguments> getDataForTask2() {
        return Stream.of(
                Arguments.of("de/havox_design/aoc2019/day18/day18part2sample.txt", 0L)
        );
    }
}

File: aoc2020/aoc2020-kotlin/src/test/kotlin/de/havox_design/aoc2020/day18/Day18Test.kt
package de.havox_design.aoc2020.day18

import de.havox_design.aoc.utils.kotlin.helpers.tests.shouldBe
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.MethodSource
import java.util.stream.Stream

class Day18Test {

    @ParameterizedTest
    @MethodSource("getDataForTestProcessPart1")
    fun testProcessPart1(filename: String, expectedResult: Long) =
        OperationOrder(filename).processPart1().shouldBe(expectedResult)

    @ParameterizedTest
    @MethodSource("getDataForTestProcessPart2")
    fun testProcessPart2(filename: String, expectedResult: Long) =
        OperationOrder(filename).processPart2().shouldBe(expectedResult)

    companion object {
        @JvmStatic
        private fun getDataForTestProcessPart1(): Stream<Arguments> =
            Stream.of(
                Arguments.of("de/havox_design/aoc2020/day18/day18part1sample1.txt", 71L),
                Arguments.of("de/havox_design/aoc2020/day18/day18part1sample2.txt", 51L),
                Arguments.of("de/havox_design/aoc2020/day18/day18part1sample3.txt", 26L),
                Arguments.of("de/havox_design/aoc2020/day18/day18part1sample4.txt", 437L),
                Arguments.of("de/havox_design/aoc2020/day18/day18part1sample5.txt", 12240L),
                Arguments.of("de/havox_design/aoc2020/day18/day18part1sample6.txt", 13632L)
            )

        @JvmStatic
        private fun getDataForTestProcessPart2(): Stream<Arguments> =
            Stream.of(
                Arguments.of("de/havox_design/aoc2020/day18/day18part2sample.txt", 0L)
            )
    }
}
